src/components/Ledger/LedgerTransportSwitchModal.js

```diff
@@ -0,0 +1,107 @@
+// @flow
+
+import React from 'react'
+import {View, ScrollView, Platform} from 'react-native'
+import {injectIntl, defineMessages, intlShape} from 'react-intl'
+
+import {Text, Button, Modal} from '../UiKit'
+
+import styles from './styles/LedgerTransportSwitchModal.style'
+
+import type {ComponentType} from 'react'
+
+const messages = defineMessages({
+  title: {
+    id: 'components.ledger.ledgertransportswitchmodal.title',
+    defaultMessage: '!!!Choose Connection Method',
+  },
+  usbExplanation: {
+    id: 'components.ledger.ledgertransportswitchmodal.usbExplanation',
+    defaultMessage:
+      '!!!Choose this option if you want to connect to a Ledger Nano model X ' +
+      'or S using an on-the-go USB cable adaptor:',
+  },
+  usbButton: {
+    id: 'components.ledger.ledgertransportswitchmodal.usbButton',
+    defaultMessage: '!!!Connect with USB',
+  },
+  usbButtonDisabled: {
+    id: 'components.ledger.ledgertransportswitchmodal.usbButtonDisabled',
+    defaultMessage: '!!!Connect with USB\n(Blocked by Apple for iOS)',
+  },
+  bluetoothExaplanation: {
+    id: 'components.ledger.ledgertransportswitchmodal.bluetoothExaplanation',
+    defaultMessage:
+      '!!!Choose this option if you want to connect to a Ledger Nano model X ' +
+      'through Bluetooth:',
+  },
+  bluetoothButton: {
+    id: 'components.ledger.ledgertransportswitchmodal.bluetoothButton',
+    defaultMessage: '!!!Connect with Bluetooth',
+  },
+})
+
+type Props = {|
+  intl: intlShape,
+  visible: boolean,
+  onSelectUSB: () => any,
+  onSelectBLE: () => any,
```

Comment:
@SebastienGllmt I know you always tell me to avoid `any` but these could be used in different contexts, so not sure if I can be more precise.